### PR TITLE
docs(maintenance): add versioning and maintenance policy

### PR DESCRIPTION
**Pull request by heitorlessa**
## Description of your changes

Formalize our versioning policy with a dedicated page similar to the AWS SDK, adjusted to Powertools for AWS Lambda (TypeScript) and our community.

### Related issues, RFCs

**Issue number:** #1995

## Checklist

- [x] [My changes meet the tenets criteria](https://docs.powertools.aws.dev/lambda/typescript/#tenets)
- [x] I have performed a *self-review* of my own code
- [x] I have *commented* my code where necessary, particularly in areas that should be flagged with a TODO, or hard-to-understand areas
- [x] I have made corresponding changes to the *documentation*
- [x] My changes generate *no new warnings*
- [x] I have *added tests* that prove my change is effective and works
- [x] The PR title follows the [conventional commit semantics](https://github.com/aws-powertools/powertools-lambda-typescript/blob/main/.github/semantic.yml#L2)

### Breaking change checklist

***Is it a breaking change?:*** NO

- [ ] I have documented the migration process
- [ ] I have added, implemented necessary warnings (if it can live side by side)

---

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

**Disclaimer**: We value your time and bandwidth. As such, any pull requests created on non-triaged issues might not be successful.

### CODE REVIEW

**Comment by heitorlessa** [comment]
@dreamorosi @am29d ready for review. 

Two recommendations after merging tho:

1. Add a documentation banner to emphasize the v2 RFC
2. Update the RFC with the target date for v2

File: docs/versioning.md
Comment (dreamorosi) [inline docs/versioning.md:58]: Just a couple minor changes in the table:
- Let's use official name convention for consistency `Powertools for AWS Lambda (TypeScript)`
- Update GA date to 07/15/2022

File: mkdocs.yml
Comment (dreamorosi) [inline mkdocs.yml:1]: Not sure if intentional but the indentation has changed for all the items

**Review by dreamorosi — CHANGES_REQUESTED** [review]
Hi @heitorlessa thank you so much for taking the time to open a PR.

Content looks good and in line with what agreed internally, so we're good on that side.

I've left 2 comments, one of them about a change.

Looking forward to merge this after they're addressed!

File: docs/versioning.md
Comment (heitorlessa) [reply docs/versioning.md:58]: On it!

Comment (heitorlessa) [reply docs/versioning.md:58]: Fixed GA date. For the naming, similar to Python, we use the `(<language>)` at the top, since that's the product umbrella. We don't have to add the `(<language>)` for every sentence as it's already lengthy. 

I've updated the info banner with the `(TypeScript)` to be triple sure there aren't confusions if the title isn't read either.

File: mkdocs.yml
Comment (heitorlessa) [reply mkdocs.yml:1]: I highly suspect is my VSCode autoformatter :/ want me to revert the mkdocs.yml?

**Review by dreamorosi — APPROVED** [review]
Thank you for this PR, as I said in the other one this will be very useful!

**Pull request merged**
